src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java

```diff
@@ -1363,6 +1368,27 @@ void testPlayerTeleport_WithoutCause_EventFired()
 		player.teleport(player.getLocation().add(10, 10, 10));
 
 		server.getPluginManager().assertEventFired(PlayerTeleportEvent.class);
+		server.getPluginManager().assertEventNotFired(EntityTeleportEvent.class);
+	}
+
+	@Test
+	void testPlayerTeleport_ChangedWorldEvent()
+	{
+		TestPlugin plugin = MockBukkit.load(TestPlugin.class);
+		World from = player.getWorld();
+		World[] playerWorldWhenEvent = new World[1];
```

Comment:
Is there a particular reason this is an array?

Reply:
The variable needs to be effectively final to be accessed from within the inner class. The array is, or it could be any holder object like an `AtomicReference` but that would be overkill.

Reply:
In fact, the assertion can be made in the inner class. This removes the array.

Reply:
Exceptions are just ignored in events so the new way can't fail. However, this is an issue with MockBukkit, so I'm going to mark this resolved and fix it in its own PR.